### ISSUE TITLE
Constraints file not used during pip install

### ISSUE THREAD

**Issue by palango** [raiden-network/raiden#3460]
The raiden-services will use Raiden as a dependency for accessing code like messages and matrix integration.

While adding this dependency a lot of warnings about incompatible package versions are shown (on a fresh venv):
```
raiden 0.2.1.dev3737+gcde27e34 has requirement click==6.7, but you'll have click 7.0 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement coincurve==8.0.2, but you'll have coincurve 11.0.0 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement colorama==0.3.9, but you'll have colorama 0.4.1 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement eth-tester==0.1.0b32, but you'll have eth-tester 0.1.0b33 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement eth-utils==1.2.1, but you'll have eth-utils 1.4.1 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement gevent==1.3.6, but you'll have gevent 1.4.0 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement ipython==4.2.1, but you'll have ipython 7.2.0 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement networkx==2.1, but you'll have networkx 2.2 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement psutil==5.4.7, but you'll have psutil 5.5.0 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement py-ecc==1.4.3, but you'll have py-ecc 1.4.7 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement py-solc==3.1.0, but you'll have py-solc 3.2.0 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement pytest==4.1.1, but you'll have pytest 4.2.0 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement requests==2.20.0, but you'll have requests 2.21.0 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement rlp==1.0.2, but you'll have rlp 1.1.0 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement structlog==18.2.0, but you'll have structlog 19.1.0 which is incompatible.
raiden 0.2.1.dev3737+gcde27e34 has requirement web3==4.7.1, but you'll have web3 4.8.2 which is incompatible.
```

It seems like the constrains from `constraints.txt` are not used when installing raiden via `pip`.

**Comment by palango** [comment]
@ulope Do you have an idea on how to fix this?

**Comment by ulope** [comment]
Hm ok, so the problem is that raiden _**is**_ using `constraints.txt` as the source for it's `setup.py` `install_requires`. This means that if the dependee (is that a word?) is using any of the same packages as raiden does it will most likely cause a conflict since pip has no dependency resolution ahead of install time. It just checks the constraints afterwards which is where those messages come from.

Solution: `¯\_(ツ)_/¯`

Either:
- Stop using constraints.txt as the install_requires source in raiden (but that has caused issues in the past due unpredictably upgraded packages, esp. from the Python Ethereum side) or 
- Downstream packages using raiden as a library need to use the exact versions raiden is using or
- Try again to use a better tool for dependency management again (probably more of a long term thing)

**Comment by palango** [comment]
> Stop using constraints.txt as the install_requires source in raiden (but that has caused issues in the past due unpredictably upgraded packages, esp. from the Python Ethereum side)

Maybe it would already help to pin the direct dependencies in `requirements.txt` and leave all other stuff to `constraints.txt`?

**Comment by LefterisJP** [comment]
> Maybe it would already help to pin the direct dependencies in `requirements.txt` and leave all other stuff to `constraints.txt`?

I also think this is a good idea for the moment. And I guess it would immediately help your use-case paul.

**Comment by ulope** [comment]
Solved with #4189
